scenario.py

```diff
@@ -633,3 +653,25 @@ def to_dataframe(self):
                 df = df.append(dict_results, ignore_index=True)
 
         return df
+
+    def shorten_dataset_proportion(self):
+        """Truncate the dataset depending on self.proportion_dataset"""
+
+        fraction = self.proportion_dataset
```

Comment:
Hmmm, when one renames a variable before using it, it oftens reveals that this variable is badly named 😃 
I suggest that you:

- rename `proportion_dataset` to `dataset_proportion`
- keep it like that in this method